pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -395,19 +400,43 @@ func (m *managerImpl) processShutdownEvent() error {
 			}(pod, group)
 		}
 
+		var groupTerminationWaitDuration = time.Duration(group.ShutdownGracePeriodSeconds) * time.Second
 		var (
-			doneCh = make(chan struct{})
-			timer  = m.clock.NewTimer(time.Duration(group.ShutdownGracePeriodSeconds) * time.Second)
+			doneCh         = make(chan struct{})
+			timer          = m.clock.NewTimer(groupTerminationWaitDuration)
+			ctx, ctxCancel = context.WithTimeout(context.Background(), groupTerminationWaitDuration)
 		)
 		go func() {
 			defer close(doneCh)
+			defer ctxCancel()
 			wg.Wait()
+
+			// The signal to kill a Pod was sent successfully to all the pods,
+			// let's wait until all the volumes are unmounted from all the pods before
+			// continuing to the next group. This is done so that the CSI Driver (assuming
+			// that it's part of the highest group) has a chance to perform unmounts.
+			if err := m.volumeManager.WaitForAllPodsUnmount(ctx, group.Pods); err != nil {
+				var podNames []string
+				for _, pod := range group.Pods {
+					podNames = append(podNames, pod.Name)
+				}
+
+				// Waiting for volume teardown is done on a best basis effort,
+				// report an error and continue.
+				//
+				// Depending on the user provided kubelet configuration value
+				// either the `timer` will tick and we'll continue to shutdown the next group, or,
+				// WaitForAllPodsUnmount will timeout, therefore this goroutine
+				// will close doneCh and we'll continue to shutdown the next group.
+				m.logger.V(1).Error(err, "Failed while waiting for all the volumes belonging to Pods in this group to unmount", "pods", podNames)
```

Comment:
do you need `V(1)` here? Just `Error` must be fine unless I am missing something

Reply:
`V(1)` is not needed here. Just `Error` is indeed fine, fixed.